src/Sentry/Internal/MainSentryEventProcessor.cs

```diff
@@ -133,6 +135,36 @@ public SentryEvent Process(SentryEvent @event)
             return @event;
         }
 
+        private void AddMemoryInfo(Contexts contexts)
+        {
+#if NETCOREAPP3_0_OR_GREATER
+            var memoryInfo = GC.GetGCMemoryInfo();
+            var values = new Dictionary<string, string>
+            {
+                {"TotalAllocatedBytes", GC.GetTotalAllocatedBytes().ToString()},
+                {"FragmentedBytes", memoryInfo.FragmentedBytes.ToString()},
+                {"HeapSizeBytes", memoryInfo.HeapSizeBytes.ToString()},
+                {"HighMemoryLoadThresholdBytes", memoryInfo.HighMemoryLoadThresholdBytes.ToString()},
+                {"TotalAvailableMemoryBytes", memoryInfo.TotalAvailableMemoryBytes.ToString()},
+                {"MemoryLoadBytes", memoryInfo.MemoryLoadBytes.ToString()},
+            };
+#if NET5_0_OR_GREATER
+            values.Add("TotalCommittedBytes", memoryInfo.TotalCommittedBytes.ToString());
+            values.Add("PromotedBytes", memoryInfo.PromotedBytes.ToString());
+            values.Add("PinnedObjectsCount", memoryInfo.PinnedObjectsCount.ToString());
+            values.Add("PauseTimePercentage", memoryInfo.PauseTimePercentage.ToString(CultureInfo.InvariantCulture));
+            var pauseDurations = memoryInfo.PauseDurations.ToArray();
+            values.Add("PauseDurations", string.Join(';', pauseDurations.Select(x => x.ToString())));
+            values.Add("Index", memoryInfo.Index.ToString());
+            values.Add("Generation", memoryInfo.Generation.ToString());
+            values.Add("FinalizationPendingCount", memoryInfo.FinalizationPendingCount.ToString());
+            values.Add("Concurrent", memoryInfo.Concurrent.ToString());
+            values.Add("Compacted", memoryInfo.Compacted.ToString());
+#endif
+            contexts[MemoryInfoKey] = values;
```

Comment:
Just realized that this will be serialized using reflection (through that "dynamic" code path).
Perhaps we should consider a class that implements `IJsonSerializable` instead